test/integration/password-authorization.int.c

```diff
@@ -106,43 +98,32 @@ test_invoke (TSS2_SYS_CONTEXT *sapi_context)
 
     TSS2_RC rval;
     int i, ret;
-
     TPM2B_MAX_NV_BUFFER nvWriteData;
-
-    char password[] = "test password";
-
-    TPMS_AUTH_COMMAND auth_command = {
-        .sessionHandle = TPM2_RS_PW,
-    };
-    TPMS_AUTH_COMMAND *auth_command_array[1] = { &auth_command, };
-    TSS2_SYS_CMD_AUTHS cmd_auths = {
-        .cmdAuthsCount = 1,
-        .cmdAuths      = auth_command_array,
-    };
-    TPMS_AUTH_RESPONSE  auth_response = { 0 };
-    TPMS_AUTH_RESPONSE *auth_response_array[1] = { &auth_response };
-    TSS2_SYS_RSP_AUTHS  rsp_auths = {
-        .rspAuths      = auth_response_array,
-        .rspAuthsCount = 1
+#define PASSWORD "test password"
+    TSS2L_SYS_AUTH_COMMAND cmd_auths = {
+        .count = 1,
+        .auths = {{
+            .sessionHandle = TPM2_RS_PW,
+            .hmac.buffer = {PASSWORD},
+            .hmac.size = strlen(PASSWORD),
+        }}
     };
 
+    TSS2L_SYS_AUTH_RESPONSE  rsp_auths = { .count = 1 };
+
     /*
      * Create an NV index that will use password
      * authorizations the password will be
      * "test password".
      */
     ret = CreatePasswordTestNV (sapi_context,
                                 TPM20_INDEX_PASSWORD_TEST,
-                                password);
+                                PASSWORD);
     if (ret != 1) {
 	// Error message already printed in CreatePasswordTestNV()
         return 1;
     }
 
-    // Init password (HMAC field in authorization structure).
-    auth_command.hmac.size = strlen (password);
-    memcpy (auth_command.hmac.buffer, password, strlen (password));
```

Comment:
Is this missing now or was it dead code to begin with ?

Reply:
It was valid when we initialized the auths in the old way.
Now this is done where the cmd_auths in defined above.

Reply:
I see